governance/program/src/state.rs

```diff
@@ -0,0 +1,219 @@
+//! Program accounts
+
+use solana_program::{epoch_schedule::Slot, instruction::Instruction, pubkey::Pubkey};
+
+/// Max number of instructions allowed for a proposal
+pub const MAX_INSTRUCTIONS: usize = 5;
+
+/// Defines all Governance accounts types
+#[derive(Clone, Debug, PartialEq)]
+pub enum GovernanceAccountType {
+    /// 0 - Default uninitialized account state
+    Uninitialized,
+
+    /// 1 - Governance account
+    Governance,
+
+    /// 2 - Proposal account for Governance account. A single Governance account can have multiple Proposal accounts
+    Proposal,
+
+    /// 3 - Proposal voting state account. Every Proposal account has exactly one ProposalState account
+    ProposalState,
+
+    /// 4 - Vote record account for a given Proposal.  Proposal can have 0..n voting records
+    VoteRecord,
+
+    /// 5 - Custom Single Signer Instruction account which holds an instruction to execute for Proposal
+    CustomSingleSignerInstruction,
```

Comment:
Wouldn't the instruction here require no signers?  If any signers are required, this means that nobody but the signer can call `Execute` to trigger the instructions once the proposal is accepted I think

Reply:
The instruction would be executed using Governance PDA as signer